cpp/CMakePresets.json

```diff
@@ -6,6 +6,30 @@
     "patch": 0
   },
   "configurePresets": [
+    {
+      "name": "_allocator-none",
+      "hidden": true,
+      "cacheVariables": {
+        "ARROW_JEMALLOC": "OFF",
+        "ARROW_MIMALLOC": "OFF"
+      }
+    },
+    {
+      "name": "_allocator-jemalloc",
+      "hidden": true,
+      "cacheVariables": {
+        "ARROW_JEMALLOC": "ON",
+        "ARROW_MIMALLOC": "OFF"
```

Comment:
Do we need this?
How about specifying only `ARROW_JEMALLOC` here?

Reply:
You mean something like this?
```suggestion
```

I think we should always specify one `ON` one `OFF` together (unless we are 100% sure about one value of them being specified earlier) - otherwise there could be potentially two allocators which is not desired?

Reply:
We can enable multiple allocators at the same time. :-)

The default allocator is chosen based on our recommendation:

https://github.com/apache/arrow/blob/5181c24f6d780591426e0af0d1b72c68f190484c/cpp/src/arrow/memory_pool.cc#L143
https://github.com/apache/arrow/blob/5181c24f6d780591426e0af0d1b72c68f190484c/cpp/src/arrow/memory_pool.cc#L86-L100

But users can select any allocator from enabled allocators by creating `arrow::MemoryPool` instead of using `arrow::default_memory_pool()`.

Reply:
Ahh, I see.

However I think we are expecting these allocator presets to enforce the allocator selection. That is, `_allocator_X` should mean "only enable allocator X, nothing else".

So for intentionally enabling both selectors, we can have another `_allocator_all` maybe?

Reply:
OK. Let's use the approach.

Reply:
But is `_allocator_all` useful enough to worth an individual cmake preset? (It makes the `_allocator` group intact though).

Reply:
Can we use it for `features-maximal`?

Reply:
Yeah that makes much more sense. Updated. Also changed to use plural `_allocators`.